### PR TITLE
Add filters to stock item transactions and batch information table

### PR DESCRIPTION
**Pull request by jabahum**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR adds filters to stock Item transactions table, batch information table and finally some improvements on the UI

## Screenshots

image

image

image

## Related Issue

https://openmrs.atlassian.net/browse/O3-3561

### CODE REVIEW

File: src/stock-items/stock-items-table.component.tsx [hunk 1]
```diff
@@ -40,6 +42,10 @@ const StockItemsTableComponent: React.FC<StockItemsTableProps> = () => {
   const { t } = useTranslation();
   const [searchInput, setSearchInput] = useState("");
 
+  const handleRefresh = () => {
+    // search.refetch()
```
Comment (makombe) [inline src/stock-items/stock-items-table.component.tsx:46]: With this commented out, means the function is doing nothing and should be removed

File: src/stock-operations/stock-operations-table.component.tsx [hunk 2]
```diff
@@ -434,6 +439,11 @@ const StockOperations: React.FC<StockOperationsTableProps> = () => {
                     onFilterChange={handleOnFilterChange}
                   />
                 </div>
+                <TableToolbarMenu>
+                  <TableToolbarAction onClick={handleRefresh}>
+                    Refresh
+                  </TableToolbarAction>
+                </TableToolbarMenu>
```
Comment (makombe) [inline src/stock-operations/stock-operations-table.component.tsx:446]: This refresh does nothing, its implementation commented out on line`70`

File: src/stock-items/stock-items-table.scss [hunk 3]
```diff
@@ -2,89 +2,97 @@
...
+  align-items: center;
+}
 
+.filterContainer {
+  display: flex;
+  align-items: center;
+}
+
+.dateAlign {
+  padding-top: 8px;
```
Comment (makombe) [inline src/stock-items/stock-items-table.scss:81]:
```suggestion
  padding-top: spacing.$spacing-03;
```

File: src/stock-items/stock-items-table.scss [hunk 4]
```diff
@@ -2,89 +2,97 @@
...
+  height: 100%;
+}
+
+.filtersAlign {
+  padding-bottom: 8px;
```
Comment (makombe) [inline src/stock-items/stock-items-table.scss:86]:
```suggestion
  padding-bottom: spacing.$spacing-03;
```

File: src/stock-items/stock-items-table.scss [hunk 5]
```diff
@@ -2,89 +2,97 @@
...
+  :global(.cds--list-box__menu) {
+    min-width: 15rem;
+    z-index: 1;
+  }
+}
+.filterContainer {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  margin-bottom: 1rem;
```
Comment (makombe) [inline src/stock-items/stock-items-table.scss:97]:
```suggestion
  margin-bottom: spacing.$spacing-05;
```

File: src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx [hunk 6]
```diff
@@ -51,7 +51,7 @@ const PackagingUnits: React.FC<PackagingUnitsProps> = ({
       },
       {
         key: "quantity",
-        header: t("quantity", "Quantity"),
+        header: t("quantity", "Pack Size"),
```
Comment (makombe) [inline src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx:54]:
```suggestion
        header: t("packSize", "Pack Size"),
```

File: src/stock-items/add-stock-item/transactions/transactions.component.tsx [hunk 7]
```diff
@@ -127,24 +145,43 @@ const Transactions: React.FC<TransactionsProps> = ({ stockItemUuid }) => {
...
+    <DataList
+      children={() => (
+        <div style={{ display: "flex", justifyContent: "space-between" }}>
+          <TransactionsLocationsFilter
+            onLocationIdChange={(q) => {
+              setLocationUuid(q);
+              setStockItemFilter({ ...stockItemFilter, locationUuid: q });
+            }}
+            name="TransactionLocationUuid"
+            placeholder="Filter by Location"
```
Comment (makombe) [inline src/stock-items/add-stock-item/transactions/transactions.component.tsx:173]: Make this translatable.

**Review by donaldkibet — APPROVED** [review]

**Review by hadijahkyampeire — APPROVED** [review]
LGTM, thanks @jabahum

**Pull request merged**